### PR TITLE
Publish docs to gh pages

### PR DESCRIPTION
**Pull request by dynobo**
See result at [boschresearch.github.io/blackboxopt](https://boschresearch.github.io/blackboxopt/).

### CODE REVIEW

**Comment by LGro** [comment]
How do you feel about going with `mkdocs-versioning` and doing `mkdocs-versioning deploy` to get https://boschresearch.github.io/blackboxopt/1.0.1 like URLs and retain documentation for older versions?

**Review by LGro — APPROVED** [review]

**Comment by dynobo** [comment]
> How do you feel about going with `mkdocs-versioning` and doing `mkdocs-versioning deploy` to get https://boschresearch.github.io/blackboxopt/1.0.1 like URLs and retain documentation for older versions?

Versioning is a good idea, but instead `mkdocs-versioning` let's go with `mike` which [seems to be the standard for mkdocs-material](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/#versioning)

File: .github/workflows/python.yaml [hunk 1]
```diff
@@ -39,12 +39,34 @@ jobs:
...
-      - name: Release
+      - name: Build & release
         env:
           PYPI_TOKEN: ${{ secrets.PYPI_TOKEN }}
         run: |
           poetry config pypi-token.pypi $PYPI_TOKEN
           poetry publish --build
+  docs:
+    needs: test
+    # if: github.ref == 'refs/heads/main'
```
Comment (LGro) [inline .github/workflows/python.yaml:53]: Do I understand correctly that for the desired state we would only run this on tagged releases?

**Review by LGro — APPROVED** [review]

**Comment by LGro** [comment]
Reminder for us to remove the commented out release condition after we have confirmed the docs deployment works.

File: .github/workflows/python.yaml (hunk 1 above)
Comment (dynobo) [reply .github/workflows/python.yaml]: As discussed, I changed to same condition as we have for pypi deployment: if tag starts with `v`.

**Pull request merged**